staging/src/k8s.io/apiserver/pkg/cel/openapi/resolver/refs.go

```diff
@@ -97,7 +99,9 @@ func populateRefs(schemaOf func(ref string) (*spec.Schema, bool), visited sets.S
 		}
 		if populated != result.Items.Schema {
 			changed = true
-			result.Items.Schema = populated
+			newItems := *result.Items
+			newItems.Schema = populated
+			result.Items = &newItems
```

Comment:
this change looks good, can you add a unit test that would have caught the issue that this mutated the original?